### ISSUE TITLE
MLM Weights

### ISSUE THREAD

**Issue by lbugnon** [CatIIIIIIII/RNAErnie#5]
Hi, Is it possible to have the pre-trained MLM weights as well? I mean the final decoder in the MLMHead. Thanks!

**Comment by CatIIIIIIII** [comment]
Hi, I think pre-trained MLM is saved and you could load it by ErnieForMaskedLM (https://paddlenlp.readthedocs.io/zh/latest/source/paddlenlp.transformers.ernie.modeling.html#paddlenlp.transformers.ernie.modeling.ErnieForMaskedLM)

**Comment by lbugnon** [comment]
Thanks! I tried that with the folowing code and the last warning seems to indicate that the weights are not being read. Im using the weights from the readme [gdrive](https://drive.google.com/drive/folders/1Ls5k7hv83BLRTznB4XcegIa2yKkU40Ls)

```import paddle
from paddlenlp.transformers import ErnieForMaskedLM

rna_ernie = ErnieForMaskedLM.from_pretrained("output/BERT,ERNIE,MOTIF,PROMPT/checkpoint_final/")
```

```[2024-09-07 21:13:18,808] [    INFO] - Loading configuration file output/BERT,ERNIE,MOTIF,PROMPT/checkpoint_final/model_config.json
[2024-09-07 21:13:18,811] [    INFO] - Loading weights file output/BERT,ERNIE,MOTIF,PROMPT/checkpoint_final/model_state.pdparams
[2024-09-07 21:13:19,259] [    INFO] - Loaded weights file from disk, setting weights to model.
[2024-09-07 21:13:20,167] [ WARNING] - Some weights of the model checkpoint at output/BERT,ERNIE,MOTIF,PROMPT/checkpoint_final/ were not used when initializing ErnieForMaskedLM: ['cls.predictions.decoder_weight']
- This IS expected if you are initializing ErnieForMaskedLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing ErnieForMaskedLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
[2024-09-07 21:13:20,170] [ WARNING] - Some weights of ErnieForMaskedLM were not initialized from the model checkpoint at output/BERT,ERNIE,MOTIF,PROMPT/checkpoint_final/ and are newly initialized: ['cls.predictions.decoder.bias', 'cls.predictions.decoder.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.```

**Comment by CatIIIIIIII** [comment]
Hi! It wotks well and has no warning on my device. Did you change the size of vocabulary? 

```
import paddle
from rna_ernie import BatchConverter
from paddlenlp.transformers import ErnieModel, ErnieForMaskedLM

# ========== Set device
# paddle.set_devickbe("gpu")

# ========== Prepare Data
data = [
    ("RNA1", "GGGUGCGAUCAUACCAGCACUAAUGCCCUCCUGGGAAGUCCUCGUGUUGCACCCCU"),
    ("RNA2", "GGGUGUCGCUCAGUUGGUAGAGUGCUUGCCUGGCAUGCAAGAAACCUUGGUUCAAUCCCCAGCACUGCA"),
    ("RNA3", "CGAUUCNCGUUCCC--CCGCCUCCA"),
]
# data = "./data/ft/seq_cls/nRC/test.fa"

# ========== Batch Converter
batch_converter = BatchConverter(k_mer=1,
                                 vocab_path="./data/vocab/vocab_1MER.txt",
                                 batch_size=256,
                                 max_seq_len=512)
# load model parameters and print all parameters
print("=====Loading model parameters...")
state_dict = paddle.load(
    "output/BERT,ERNIE,MOTIF,PROMPT/checkpoint_final/model_state.pdparams")
for name, param in state_dict.items():
    print(name)
# init a new model for

# ========== RNAErnie Model
print("\n=====Initializing model...")
rna_ernie = ErnieForMaskedLM.from_pretrained(
    "output/BERT,ERNIE,MOTIF,PROMPT/checkpoint_final/")
# print parameters that has been loaded
for name, param in rna_ernie.named_parameters():
    print(name)

```

Output:

```
=====Loading model parameters...
ernie.embeddings.word_embeddings.weight
ernie.embeddings.position_embeddings.weight
ernie.embeddings.token_type_embeddings.weight
ernie.embeddings.layer_norm.weight
ernie.embeddings.layer_norm.bias
ernie.encoder.layers.0.self_attn.q_proj.weight
...
ernie.pooler.dense.weight
ernie.pooler.dense.bias
cls.predictions.decoder_weight
cls.predictions.decoder_bias
cls.predictions.transform.weight
cls.predictions.transform.bias
cls.predictions.layer_norm.weight
cls.predictions.layer_norm.bias

=====Initializing model...
ernie.embeddings.word_embeddings.weight
ernie.embeddings.position_embeddings.weight
ernie.embeddings.token_type_embeddings.weight
ernie.embeddings.layer_norm.weight
ernie.embeddings.layer_norm.bias
ernie.encoder.layers.0.self_attn.q_proj.weight
...
ernie.pooler.dense.weight
ernie.pooler.dense.bias
cls.predictions.decoder_bias
cls.predictions.transform.weight
cls.predictions.transform.bias
cls.predictions.layer_norm.weight
cls.predictions.layer_norm.bias
```

**Comment by lbugnon** [comment]
mmm I'm still geting the warnings, but when using ErnieForMaskedLM to get token logits I get reasonable token predictions for well-known sequences (like tRNA), so indeed the MLM model is being loaded and working. 

Maybe is a paddle version thing? I use a colab and install the default versions of paddlepaddle-gpu and paddlenlp as repo instructions didnt work for me

**Comment by CatIIIIIIII** [comment]
Yes, that maybe the problem. I have to admit that paddle did not manage different versions very well.

And if you failed to install paddlepaddle from instruction, you could install it following the official website. 
_paddlenlp==2.4.0
paddlepaddle-gpu==2.3.2_
The conda file provided uses tsinghua source which is fast for China mainland users, but may cause inconvenience for other users.

Moreover, I also pretrained a advanced version with pytorch. This version support up to 1024 length sequences and use the newest version of RNAcentral with 31 million sequences. You could find it here [model](https://huggingface.co/WANGNingroci/RNAErnie), [code](https://github.com/CatIIIIIIII/RNAErnie2). 

Also, [multimolecule](https://github.com/DLS5-Omics/multimolecule) provides an efficient way to load all the related RNA foundation models including RNAErnie.
